src/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+
+export const useLocalStorage = <T extends string>(
+  key: string,
+  initialValue: T,
+): [T, (value: T | ((val: T) => T)) => void] => {
+  const [state, setState] = useState<T>(() => {
+    if (typeof window !== "undefined") {
```

Comment:
this is the important bit, checks for the window before running and of the logic

Reply:
Right, we temporarily added this if statement to the theme-provider.tsx last time we talked.